### ISSUE TITLE
no-map-without-usage mistakenly trigged on arrow function returns

### ISSUE THREAD

**Issue by KurtPreston** [Glavin001/tslint-clean-code#67]
The following line should be valid:

```typescript
const myFunc = () => (
   [].map((x) => x)
);
```

However, `no-map-without-usage` cannot detect that a value is being returned.
